### ISSUE TITLE
Filter Support in Subscription (rest-api)

### ISSUE THREAD

**Issue by gauravAshok** [flipkart-incubator/varadhi#124]

**Comment by kmrdhruv** [comment]
Instead of restricting the nesting to one level, let the implementation be open for multi level nesting. Restrictions can be added via config at parsing level to restrict the depth of nesting.

**Comment by gauravAshok** [comment]
> Instead of restricting the nesting to one level, let the implementation be open for multi level nesting. Restrictions can be added via config at parsing level to restrict the depth of nesting.

Yes, the model is not restrictive, and will be able to express most expectations.

**Comment by kmrdhruv** [comment]
for condition struct, can we evaluate "exists" operator as well i.e.checks for presence of header, not its value.

**Comment by AayuStark007** [comment]
I'm assuming, that the `exp` field in Expression Struct can contain a list of Expression Structs or Condition Structs. Is that correct?

**Comment by gaurav-ashok** [comment]
> I'm assuming, that the `exp` field in Expression Struct can contain a list of Expression Structs or Condition Structs. Is that correct?

Yes. or the mix of both. to represent a scenario of "exp": (C1 AND C2) OR C3. where "C1 AND C2" itself is a expression containing 2 conditions. C3 is a standalone condition.
